### PR TITLE
[WIP] Sort .cvs export

### PR DESCRIPTION
**Pull request by bramleyjl**
Changes the .csv export file to sort by Color -> CMC -> Alphabetical

### CODE REVIEW

File: src/utils/Sort.js [hunk 1]
```diff
@@ -727,3 +727,16 @@ export function countGroup(group) {
   }
   return group.length;
 }
+
+export function downloadSort(cards) {
```
Comment (dekkerglen) [inline src/utils/Sort.js]: Can you rename this to sortForCSVDownload? downloadSort sounds like it should download a sort.

File: src/utils/Sort.js [hunk 2]
```diff
@@ -727,3 +727,16 @@ export function countGroup(group) {
...
+  var exportCards = [];
+  cards = sortDeep(cards, 'Color', 'CMC');
+  cards.forEach((firstGroup) => {
```
Comment (dekkerglen) [inline src/utils/Sort.js]: can you use for(const .. of ..) for this instead of .forEach()? This is a style change we've recently been moving towards

**Comment by dekkerglen** [comment]
Passing the current selected sorts world be very nice

**Comment by bramleyjl** [comment]
Got a bit stuck on how I'm supposed to set the primary and secondary sort values in the proper scope from my callback function; there doesn't seem to be a state to `SortCollapse` that I can set?

File: src/utils/Sort.js [hunk 3]
```diff
@@ -727,3 +727,16 @@ export function countGroup(group) {
   }
   return group.length;
 }
+
+export function sortForCSVDownload(cards) {
+  var exportCards = [];
+  cards = sortDeep(cards, 'Color', 'CMC');
+  for (const firstGroup of cards) {
+    for (const secondGroup of firstGroup[1]) {
```
Comment (dekkerglen) [inline src/utils/Sort.js]: This doesn't look right... we are we doing `firstGroup[1]`?

Comment (bramleyjl) [reply src/utils/Sort.js]: Each group produced by `sortDeep` is an array where the 0 index is the category (like `White` or `3`) and the 1 index is a subarray with either second groups or the cards themselves.

Comment (dekkerglen) [reply src/utils/Sort.js]: Right, because that was used to write the headers. Looks good to me then

**Pull request merged**